### DISCUSSION TITLE
About the AUTHORS file

### DISCUSSION THREAD

**Discussion by jorgepiloto** [ansys/pyansys-dev-guide#250] [Ideas]
One of the special files that GitHub allows to place in a repository is the `AUTHORS` file. Its content is usually written in Markdown and contains references to the legal authors of the project.

So far, we have been using the following metadata in our Python project files:

```toml
authors = [
    {name = "ANSYS, Inc.", email = "pyansys.support@ansys.com"},
]
maintainers = [
    {name = "PyAnsys developers", email = "pyansys.maintainers@ansys.com"},
]
```

Hence, the only author is always `ANSYS, Inc.` whereas the usual contributors are the maintainers of the project.

This got me thinking if we should have two separate files, one for `AUTHORS` and another one for `CONTRIBUTORS` or `MAINTAINERS`. Many softwares out there assume that authors and maintainers are the same but I bet there are differences.

**What other projects are doing**

From the [GNU project guidelines](https://www.gnu.org/prep/maintain/html_node/Recording-Contributors.html):

> Only the contributions that are legally significant for copyright purposes (see [Legally Significant](https://www.gnu.org/prep/maintain/html_node/Legally-Significant.html#Legally-Significant)) need to be listed. Small contributions, bug reports, ideas, etc., can be omitted.

For the case of Google, a single file named "AUTHORS" is used. If we agree on having a single file, then `ANSYS, Inc.` should be always placed at the top of this file, [same as what Google is doing](https://opensource.google/documentation/reference/releasing/authors).

**Comment by RobPasMue** [comment]
I think the point brought up by @phxnsharp in #248 is really interesting. Contributors come and go. Even the main maintainers of a project. IMO each project should have a maintainers team. And that team should be the author of the package (together with ``ANSYS, Inc.``). This provides a better maintainability, since you can easily add/remove people from the group without having to modify the reference (i.e. the AUTHORS file).
